File: android/src/oldarch/ReactNativeAudioSpec.java
package com.drpogodin.reactnativeaudio;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.Promise;
import com.facebook.react.modules.core.DeviceEventManagerModule;

import java.util.Map;

abstract class ReactNativeAudioSpec extends ReactContextBaseJavaModule {
  ReactNativeAudioSpec(ReactApplicationContext context) {
    super(context);
  }

  public abstract String getName();
  public abstract void configAudioSystem(Promise promise);
  public abstract Map<String,Object> getTypedExportedConstants();

  public Map<String,Object> getConstants() {
    return this.getTypedExportedConstants();
  }

  public abstract void getInputAvailable(Promise promise);

  public abstract void listen(
    double audioSource,
    double sampleRate,
    double channelConfig,
    double audioFormat,
    double samplingSize,
    Promise promise
  );

  public abstract void muteInputStream(double streamId, boolean muted);
  public abstract void unlisten(double streamId);
  public abstract void addListener(String eventName);
  public abstract void removeListeners(double count);
}

File: android/src/oldarch/ReactNativeAudioSpec.kt
package com.drpogodin.reactnativeaudio

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.Promise

abstract class ReactNativeAudioSpec(context: ReactApplicationContext?) : ReactContextBaseJavaModule(context) {
    abstract override fun getName(): String
    abstract fun configAudioSystem(promise: Promise)
    abstract fun getTypedExportedConstants(): MutableMap<String, Any>
    abstract fun destroySamplePlayer(playerId: Double, promise: Promise)
    override fun getConstants(): Map<String, Any>? {
        return getTypedExportedConstants()
    }

    abstract fun getInputAvailable(promise: Promise)
    abstract fun initSamplePlayer(playerId: Double, promise: Promise)
    abstract fun listen(
            streamId: Double,
            audioSource: Double,
            sampleRate: Double,
            channelConfig: Double,
            audioFormat: Double,
            samplingSize: Double,
            promise: Promise
    )
    abstract fun loadSample(
      playerId: Double,
      sampleName: String,
      samplePath: String,
      promise: Promise
    )
    abstract fun playSample(
      playerId: Double,
      sampleName: String,
      loop: Boolean,
      promise: Promise
    )
    abstract fun muteInputStream(streamId: Double, muted: Boolean)
    abstract fun unlisten(streamId: Double, promise: Promise)
    abstract fun addListener(eventName: String?)
    abstract fun removeListeners(count: Double)
    abstract fun stopSample(playerId: Double, sampleName: String, promise: Promise)
    abstract fun unloadSample(playerId: Double, sampleName: String, promise: Promise)
  }
